cpp/src/gandiva/date_utils.h

```diff
@@ -47,6 +58,38 @@ class DateUtils {
   static std::vector<std::string> GetExactMatches(const std::string& pattern);
 };
 
+namespace internal {
+
+/// \brief Returns seconds since the UNIX epoch
+static inline bool ParseTimestamp(const char* buf, const char* format, int64_t* out) {
```

Comment:
This will not work with MinGW.
I'll work on this later when we support Gandiva with MinGW.